xpra/server/auth/keycloak.py

```diff
@@ -158,9 +164,37 @@ def check(self, response_json: bytes) -> bool:
                 log.error("Error: keycloak authentication failed as token state is invalid")
                 return False
 
+            log("keycloak: token is active")
+
             user_info = keycloak_openid.userinfo(access_token)
-            log("userinfo_info: %r", user_info)
-            log("keycloak authentication succeeded: token is active")
+            log("user_info: %r", user_info)
+
+            log("claim_field: %r", self.claim_field)
+            log("auth_groups: %r", self.auth_groups)
+            # if claim_field is defined, check that the auth_groups match the groups_claim condition
+            if self.claim_field:
+                if not self.auth_groups:
+                    log.error("Error: keycloak authentication failed as auth_groups is invalid")
+                    return False
+
+                def get(data_dict, keys):
+                    return get(data_dict[keys.pop(0)], keys) if len(keys) else data_dict
+
+                groups_claim = get(user_info, self.claim_field.split('.'))
+                log("groups_claim: %r", groups_claim)
```

Comment:
`groups_claim` is always used as a `set(groups_claim)` so perhaps this should be done here, only once.
What type of value does `get` return? Is it guaranteed to always return an `Iterable`?

Reply:
It will return a list of groups.